tools/gadget3_to_eagleHDF.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import os
+import argparse
+from scipy.io import FortranFile
+import h5py as h5
+
+"""
+Script for converting .gadget3 binary ICs to the hdf5 format, for use with the EAGLE code.
+
+Adapted from an IDL script by Rob Crain, by Jonathan Davies, 2021.
+
+Usage: ics2hdf.py ics_file
+* Creates an hdf5 file in the same directory as the original ICs, with the same name.
+"""
+
+def ics2hdf(ics_file):
+
+    raw_ics = os.path.abspath(ics_file)
+
+    # Do quick checks to make sure input is valid
+    assert os.path.isfile(raw_ics),'Input ICs file does not exist.'
+    assert raw_ics.split('.')[-1] == 'gadget3','Input ICs not valid. Please input GADGET3 binary ICs (.gadget3)'
+
+    out_ics = raw_ics[:-7] + 'hdf5'
+
+    print('Reading binary file '+raw_ics)
+
+    # Open the FORTRAN unformatted binary ICs
+    f = FortranFile(raw_ics, 'r')
```

Comment:
Assuming you use the trick bellow, this'll have to be changed into
```suggestion
def ics2hdf(ics_file):
    assert ics_file.name.split('.')[-1] == 'gadget3', 'Input ICs not valid. Please input GADGET3 binary ICs (.gadget3)'
    
    out_ics = raw_ics.name[:-7] + "hdf5"

    print(f"Reading binary file {raw_ics}")

    # Open the FORTRAN unformatted binary ICs
    f = FortranFile(ics_file, 'r')
```